### PR TITLE
fix(modal): Remove overflow class when modal is removed from DOM

### LINKED ISSUE #1394: Bug: Calcite Modal does not remove overflow-hidden class on HTML if it removed from the DOM.
### Summary

If a open `<calcite-modal>` is removed from the DOM without being closed the `overflow-hidden` class persists on the `<html>` element.

### Actual Behavior

The  `overflow-hidden`  class persists on the `<html>` element.

### Expected Behavior

Removing a `<calcite-modal>` from the DOM should clean up after itself by removing the `overflow-hidden` class.

### Reproduction Steps

1. Open https://codepen.io/patrickarlt/pen/eYddLMp?editors=0100
1. Notice you can scroll the page because of the really long div.
1. Open the modal
1. Click the remove buttom
1. You cannot scroll the page because `overflow-hidden` remains on the body.

### Relevant Info

This kind of cleanup behavior is probably needed across any component that modifies the DOM outside itself. I ran into this when doing somethign like this in React:

```jsx
{option ? <ModalThatTogglesOption /> : <OtherModalThatTogglesOption />}
```

`<ModalThatTogglesOption>` and `<OtherModalThatTogglesOption>`  both modify the value of `option` and close then selves but React is doing this:

1. `option=true` so React renders `<ModalThatTogglesOption />`
2. A user uses `<ModalThatTogglesOption />` to make `option=false`. This occurs in a function like:
   ```
   updateOptionTo(false); <-- React re-renders here
   closeModal(); <-- this doesn't get called because the modal no longer exists in React
   ```
3. Now `option=false` React starts to re-render because closing a modal is async it goes on the stack while React is re-rendering.
4. `option=false` so React renders  `<OtherModalThatTogglesOption>` and REMOVES `<ModalThatTogglesOption />` which makes it look like the modal closed (since it removes the open modal from the DOM) but since the modal didn't close properly the `overflow-hidden`  class persists on the `<html>` element.

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** Fixes #1394 cc @patrickarlt 

## Summary
Removes the overflow class from the document on `disconnectedCallback()`

Would like to add some tests to this but having trouble accessing the document classList in the e2e test... open to suggestions there.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
Nice

File: src/components/calcite-modal/calcite-modal.tsx [hunk 1]
```diff
@@ -98,6 +98,10 @@ export class CalciteModal {
     }
   }
 
+  disconnectedCallback(): void {
+    this.removeOverflowHiddenClass;
```
Comment (paulcpederson) [inline src/components/calcite-modal/calcite-modal.tsx:102]: Don't you have to call this like a method?

Comment (driskull) [reply src/components/calcite-modal/calcite-modal.tsx:102]: ah yeah you do.

Also, can you add a test for this?

File: src/components/calcite-modal/calcite-modal.tsx [hunk 2]
```diff
@@ -311,4 +315,8 @@ export class CalciteModal {
       focusElement(this.closeButtonEl);
     }
   };
+
+  private removeOverflowHiddenClass(): void {
+    document.documentElement.classList.add("overflow-hidden");
```
Comment (paulcpederson) [inline src/components/calcite-modal/calcite-modal.tsx:320]: also, aren't you adding here, instead of removing?

Comment (paulcpederson) [reply src/components/calcite-modal/calcite-modal.tsx:320]: probably a good idea to use some optional chaining here for environments like prerender that may not have a document:

```
document?.documentElement?.classList.add("overflow-hidden")
```

**Review by driskull — CHANGES_REQUESTED** [review]
@macandcheese can you add a test and fix @paulcpederson's comments?

**Comment by driskull** [comment]
Any other components that may not be properly doing disconnected cleanup?

File: src/components/calcite-modal/calcite-modal.tsx (hunk 2 above)
Comment (macandcheese) [reply src/components/calcite-modal/calcite-modal.tsx:320]: Ha, yeah...

**Comment by paulcpederson** [comment]
@macandcheese maybe `page.evaluate` could work here? 

```
      await page.evaluate(() => {
        return document.documentElement.classList;
      });
```

**Comment by macandcheese** [comment]
> Any other components that may not be properly doing disconnected cleanup?

I believe this is the only component that set a class on document... or would have any straggler containers or things like that (🤞 ).

**Comment by macandcheese** [comment]
> @macandcheese maybe `page.evaluate` could work here?

This did work although I could only get it working by testing object equality - using "not/toHaveClass" on documentElement wasn't working.

**Review by driskull — APPROVED** [review]
👏

**Comment by paulcpederson** [comment]
Maybe like 

```
    const documentClass = await page.evaluate(() => {
      return document.documentElement.classList.contains("overflow-hidden");
    });
    expect(documentClass).toEqual(false);
```

or something?

**Comment by macandcheese** [comment]
@paulcpederson that syntax worked! But looks like there are now some date component failures after pulling master 😦

**Comment by paulcpederson** [comment]
@macandcheese probably from the date-range stuff we put in recently? Maybe we have a brittle test in that suite somewhere...

**Comment by macandcheese** [comment]
Merging in master fixed the unrelated test failures - merging.

**Pull request merged**